### PR TITLE
Update ACL during job reset

### LINKED ISSUE #38721: CreateJobOperator does not udpdate ACL
### Apache Airflow Provider(s)

databricks

### Versions of Apache Airflow Providers

apache-airflow-providers-common-io==1.2.0
apache-airflow-providers-common-sql==1.10.0
apache-airflow-providers-databricks==6.1.0
apache-airflow-providers-ftp==3.7.0
apache-airflow-providers-http==4.8.0
apache-airflow-providers-imap==3.5.0
apache-airflow-providers-sqlite==3.7.0

### Apache Airflow version

2.8.1

### Operating System

ubuntu

### Deployment

Google Cloud Composer

### Deployment details

_No response_

### What happened

DatabricksCreateJobsOperator does not update ACL list.

### What you think should happen instead

DatabricksCreateJobsOperator should update all components of job including ACLs.

### How to reproduce

Create a job config with access control list.
```
 access_control = [{
            "group_name": "abc",
            "permission_level": "CAN_MANAGE_RUN"
        }]
 ```
In the first run CreateJobOperator updates ACL properly. Now update ACL list and run CreateJobOperator again.
```
 access_control = [{
            "group_name": "abc",
            "permission_level": "CAN_MANAGE_RUN"
        },
        {
            "group_name": "def",
            "permission_level": "CAN_MANAGE_RUN"
        }
]
 ```

new ACL list does not get reflected in databricks jobs. This may be due to reset job API does not have ACL list paramater. And there is a separate API to handle ACL update.
reset job api: [link](https://docs.databricks.com/api/workspace/jobs/reset)
update ACL api: [link](https://docs.databricks.com/api/workspace/jobs/updatepermissions)

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by SubhamSinghal**
closes: https://github.com/apache/airflow/issues/38721

### CODE REVIEW

**Comment by SubhamSinghal** [comment]
@Lee-W Please review changes.

**Comment by subham611** [comment]
@Lee-W `Wait for CI images ` step is failing. How should I fix this, or simply rerun would work?

**Comment by Lee-W** [comment]
> @Lee-W `Wait for CI images ` step is failing. How should I fix this, or simply rerun would work?

Let me trigger rerun and see how it works

**Comment by subham611** [comment]
@Lee-W How to fix `Wait for CI images`  error. Need help here.

**Comment by potiuk** [comment]
Rebased it ->  your PR was 62 commits behind -> next time you can do it yourself too (see our contributing docs and search for rebase) https://github.com/apache/airflow/blob/main/contributing-docs/10_working_with_git.rst

**Comment by Lee-W** [comment]
Thanks @potiuk for help out!

@SubhamSinghal There's one static check failure. We might need your help to fix it

File: airflow/providers/databricks/hooks/databricks.py [hunk 1]
```diff
@@ -655,6 +656,16 @@ def get_repo_by_path(self, path: str) -> str | None:
 
         return None
 
+    def update_job_permission(self, json: dict[str, Any]) -> dict:
+        """
+        Update databricks job permission
+
+        :param json: payload
+        :return:
```
Comment (Lee-W) [inline airflow/providers/databricks/hooks/databricks.py]: Is the content missing?

**Review by Lee-W — APPROVED** [review]
Overall LGTM.

File: airflow/providers/databricks/hooks/databricks.py (hunk 1 above)
Comment (subham611) [reply airflow/providers/databricks/hooks/databricks.py]: added.

File: airflow/providers/databricks/operators/databricks.py [hunk 2]
```diff
@@ -316,6 +316,10 @@ def execute(self, context: Context) -> int:
         if job_id is None:
             return self._hook.create_job(self.json)
         self._hook.reset_job(str(job_id), self.json)
+        if "access_control_list" in self.json and self.json["access_control_list"] is not None:
```
Comment (pankajkoti) [inline airflow/providers/databricks/operators/databricks.py]:
```suggestion
        if self.json.get("access_control_list") is not None:
```

**Review by pankajkoti — APPROVED** [review]
LGTM. One minor suggestion inline to optimise the number of conditional checks.

File: airflow/providers/databricks/operators/databricks.py (hunk 2 above)
Comment (Lee-W) [reply airflow/providers/databricks/operators/databricks.py]: Yep, i think this is better 👍  but I noticed this practice is used elsewhere. Should we change all of them into this style? (maybe create an issue?)

Comment (Taragolis) [reply airflow/providers/databricks/operators/databricks.py]: Python 3.8+ there 😄 

```python
if (access_control_list := self.json.get("access_control_list")) is not None:
    acl_json = {"access_control_list": access_control_list}
    self._hook.update_job_permission(normalise_json_content(acl_json))
```

Comment (Lee-W) [reply airflow/providers/databricks/operators/databricks.py]: It's indeed even better 👍

**Review by pankajkoti — APPROVED** [review]

**Pull request merged**